ext/lib/mgmt/mcumgr/samples/smp_svr/zephyr/src/main.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include <zephyr.h>
+#include <string.h>
+#include <stdlib.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt.h>
+#include <stats.h>
+#include <mgmt/smp_bt.h>
+#include <mgmt/buf.h>
+
+#ifdef CONFIG_MCUMGR_CMD_FS_MGMT
+#include "fs_mgmt/fs_mgmt.h"
+#endif
+#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
+#include "os_mgmt/os_mgmt.h"
+#endif
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include "img_mgmt/img_mgmt.h"
+#endif
+#ifdef CONFIG_MCUMGR_CMD_STAT_MGMT
+#include "stat_mgmt/stat_mgmt.h"
+#endif
+
+#define DEVICE_NAME         CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN     (sizeof(DEVICE_NAME) - 1)
+
+/* Define an example stats group; tracks seconds of uptime. */
```

Comment:
This does not track seconds of uptime.

It tracks the number of times that the main thread has requested from the kernel that it be scheduled after idling for 1 second.

This could continuously return 0 seconds of uptime if the main thread is CPU-starved by higher-priority threads.

Samples that try to display uptime to the user should use k_uptime_get();

Perhaps there is some other statistic that could be used to demonstrate stat_mgmt? Or perhaps this sample could be written in such a way that mcumgr can retrieve the uptime using stat_mgmt in a canonical/correct way.

Reply:
I am hesitant to use NFFS for the example stats group because NFFS may not be present in the application.  I have added a stripped down `prj.conf` file for more constrained BSPs (e.g., nRF51), and it does not include NFFS support.

Since the purpose of this stat is to serve as an example of a statistics group, rather than provide an exact uptime measurement, I think the best way forward is to just modify the comment.  In the mcumgr repo, I have changed the comment to the following:

```
/* Define an example stats group; approximates seconds since boot. */
```

I hope this doesn't seem to weasely.  Please let me know if you think more should be done here.

Reply:
"I am hesitant to use NFFS for the example stats group because NFFS may not be present in the application. I have added a stripped down prj.conf file for more constrained BSPs (e.g., nRF51), and it does not include NFFS support."
Makes sense.

"I hope this doesn't seem to weasely. Please let me know if you think more should be done here."

This is OK.